contributor-docs/updating-components.md

```diff
@@ -0,0 +1,45 @@
+# Updating components
+
+## Deprecation and removal process
+
+Our approach to making API changes and deprecations is heavily guided by a component’s usage in the [github.com codebase](https://github.com/github/github) and your confidence in the scope of your changes. If making a breaking change to a component’s API will require none or only a few updates in github.com then it is okay to release the breaking change and include any required updates in the pull request for upgrading the Primer ViewComponents library in github.com (we haven’t established a process for communicating these yet but make sure they’re detailed in your pull request).
+
+If the change requires a lot of updates or you’re unsure of the consequences of the change it’s best to follow the deprecation process:
+
+1. Make the changes without effecting existing functionality
+
+   For significant changes consider the [component lifecycle](https://primer.style/contribute/component-lifecycle) criteria; if the changes qualify the component for the next maturity level it’s much easier to introduce, for example, a new Beta component and deprecate the Alpha component than maintaining both APIs in a single component. This also makes it easier to identify which cases have been migrated.
+
+2. Mark old functionality as deprecated
+
+   For arguments add `DEPRECATED` before the YARD argument description, for components update the `status` value to `:deprecated`.
+
+3. Write an ERB linter to detect and fix uses of old functionality
+
+   This will make it easier to find uses of the old functionality in github.com. Our ERB linters also have autocorrection logic so we can automate migrations where possible. [See our linting guide for details on how to write linters](https://primer.style/view-components/linting).
+
+4. Make a release and upgrade the Primer ViewComponents library in github.com
+5. Migrate uses in github.com to new functionality
+
+   [How do we run the migrations / get data from the linter?]
```

Comment:
@manuelpuyol can you write a bit up on this? I remember going through it in a call before xmas but can't find my notes :(

Reply:
You can either enable your linter globally adding it to the [`.erb-lint.yml`](https://github.com/github/github/blob/master/.erb-lint.yml) file and running `bin/bundle exec erblint -a app/views app/components app/packages`

or you can run only your linter with `bin/bundle exec erblint --enable-linters your_linter_name -a app/views app/components app/packages`